### PR TITLE
LDAP auth group filtering: support for $userdn

### PR DESCRIPTION
**Pull request by epol**
In the LDAP authentication module the group filtering can use the $userdn variable that contains the user DN.
This feature is useful in a Active Directory environment where the groups contain the user DNs as members.

### CODE REVIEW

**Comment by Shnoulle** [comment]
And next time ? We add more and more var ?

Personally : i don't want to _improve_ LDAPAuth like ths because

1. It's not a global system _Allow any filter_
2. LDAPAuth can be easily cloned and update for each situation purpose.

**Comment by epol** [comment]
Hi Denis and thank you for your reply.

**tl;dr:** I think that many users are using Active Directory as the authentication and authorization method, so merging this edit would make their life easier configuring LimeSurvey.

**Long answer**
I agree with you that adding a lot of vars to cover every possible use would be clumsy and potentially harmful on the performance point of view (you may have to do more LDAP queries).
But, on the other hand, I think that my extension covers enough use cases to deserve some thinking. Indeed I think that there are a lot of users using Active Directory as their LDAP database and they may want to use groups (where members are pointed using the DN); adding the _$userdn_ attribute doesn't add any additional LDAP query (the variable is already available from a previous query), so the effect on performances should be negligible.
